### PR TITLE
Implementation of both UI and back-end for the customer login flow

### LINKED ISSUE #5: As a shopper, I should be able to sign up and login to my own unique account (which is different from a store owner) so that I can save all the data that I have entered for when I place orders.

### PR DESCRIPTION
**Pull request by danielalyoshin**
Implementation of both UI and back-end for the customer login flow.

Includes:
- Completed UI
- Account creation
- Initialization of user data in Firebase
- User authentication

### CODE REVIEW

File: app/src/main/res/layout/activity_customer_login.xml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        android:textSize="16sp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/logo_customer_login_activity"
+        app:layout_constraintVertical_bias="0.76" />
+
+    <Button
+        android:id="@+id/customer_login_activity_create_account_button"
+        android:layout_width="150dp"
```
Comment (wobertt) [inline app/src/main/res/layout/activity_customer_login.xml]: "Create Account" text clips on default virtual device: https://i.snipboard.io/ze8ZtQ.jpg. Increasing width to 200dp seems to fix it.

File: app/src/main/java/com/example/b07_project_team1/CustomerCreateAccount.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    }
+                });
+        }
+
+
+    }
+}
+
+/*
+        ~DATABASE STRUCTURE~
```
Comment (wobertt) [inline app/src/main/java/com/example/b07_project_team1/CustomerCreateAccount.java]: I think we should leave the database structure comments out

File: app/src/main/res/layout/activity_customer_create_account.xml [hunk 3]
```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/pure_white"
+    tools:context=".CustomerCreateAccount">
+
+    <TextView
```
Comment (wobertt) [inline app/src/main/res/layout/activity_customer_create_account.xml:10]: Long message clips on default virtual device: https://i.snipboard.io/w75X0O.jpg. Not sure how to fix - I can't find the error message in the codebase.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+    <string name="login_button_text">Login</string>
+    <string name="email_entry_box_text">Email:</string>
+    <string name="password_entry_box_text">Password:</string>
+    <string name="customer_login_activity_invalid_credentials_error_text"></string>
+    <string name="no_account_question_text">Don\'t have an account?</string>
+    <string name="create_account_button_text_underlined"><u>Create Account</u></string>
+    <string name="repeat_password_entry_box_text">Repeat Password:</string>
+    <string name="create_account_button_text">Create Account</string>
+    <string name="create_account_empty_fields_text">You have empty fields.</string>
+    <string name="create_account_password_repeat_no_match_text">Passwords do not Match.</string>
```
Comment (wobertt) [inline app/src/main/res/values/strings.xml]: Match -> match

**Review by wobertt — CHANGES_REQUESTED** [review]
Found a few UI issues, but I'm loving the design overall

**Review by wobertt — APPROVED** [review]

**Comment by ThunderRoar** [comment]
I would recommend having the error disappear after a few seconds or having a highlight the error if they make a mistake again. For example, when I tried to input a short password and then try a bit larger one, it would not notify the user that the password still needs to be changed. So thats something we should try changing how the error format would notify.

But overall the app works great.

**Review by ThunderRoar — APPROVED** [review]

**Comment by jang-35** [comment]
Everything looks good to me, the app is functional and looks great.

**Review by jang-35 — APPROVED** [review]

**Pull request merged**